src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -197,8 +197,11 @@ public Task Stop()
 
                             await ProduceEnd();
 
-                            // Finish reading the request body in case the app did not.
-                            await RequestBody.CopyToAsync(Stream.Null);
+                            if (!MessageBody.LocalIntakeFin)
```

Comment:
Even in a non-keep-alive request you should gracefully drain the request body. This helps avoid backlog conditions on the client.